README.md

```diff
@@ -5,45 +5,164 @@
+**Components are written in a way that allows to define input and output keys, to be read and written from the nested dict**  
+See a short introduction video (3 minutes) to how FuseMedML components work:  
 
-FuseMedML is an outstanding collaboration framework that allows you to rerun an experiment or reuse some of the capabilities originally written for different projects—all with minimal effort.\
-Using FuseMedML, you can write generic components that can be easily shared between projects in a plug & play manner, simplifying sharing and collaboration.
+https://user-images.githubusercontent.com/7043815/177197158-d3ea0736-629e-4dcb-bd5e-666993fbcfa2.mp4
 
-The framework’s unique software design provides many advantages, making it an ideal framework for deep-learning research and development in medical imaging:
 
-* **Rapid development** -
+### Examples - using FuseMedML-style components
 
-* **Flexible, customizable, and scalable** -
 
-* **Encourage sharing and collaboration** - 
+A multi head model FuseMedML style component, allows easy reuse across projects:
 
-* **Collection of off-the-shelf components and capabilities** - 
+```python
+ModelMultiHead(    
+    conv_inputs=(('data.input.img', 1),),                                       # input to the backbone model   
+    backbone=BackboneResnet3D(in_channels=1),                                   # PyTorch nn Module    
+    heads=[                                                                     # list of heads - gives the option to support multi task / multi head approach
+               Head3DClassifier(head_name='classification',                     
+                                conv_inputs=[("model.backbone_features", 512)]  # Input to the classification head
+                                ,),                                              
+          ]
+)
+```
 
-* **Standardized evaluation** - 
+Our default loss implementation - creates an easy wrap around a callable function, while being FuseMedML style
+```python
+LossDefault(
+    pred='model.logits.classification',          # input - model prediction scores
+    target='data.label',                         # input - ground truth labels
+    callable=torch.nn.functional.cross_entropy   # callable - function that will get the prediction scores and labels extracted from batch_dict and compute the loss
+)
+```
 
-* **Medical imaging expertise** - 
+An example metric that can be used
+```python
+MetricAUCROC(
+    pred='model.output', # input - model prediction scores
+    target='data.label'  # input - ground truth labels
+)
+```
 
-* **Compatibility with alternative frameworks**
+Note that several components return answers directly and not write it into the nested dictionary. This is perfectly fine, and to allow maximum flexibility we do not require any usage of output path keys.
```

Comment:
Do we need this? In any event, readers don't have the understanding to parse this statement yet. Return? We haven't seen any returns or outputs yet at all. Maybe this can come after the nested dict if the nested dict is pushed lower in the sequence.
Even better, just include this option as part of a live example, in the spirit of "show don't tell".